File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    
    a = [] #24,25,26,27
    b = []
    c = 0
    
    for i in range(left, right+1):
        a.append(i)
    
    for j in range(0, len(a)): #j = 0,1,2,3,4 수의 갯수만큼 반복
        for k in range(1, a[j]+1): #k = 각 숫자 24 25 26 27 을 0부터 반복
            if(a[j] % k == 0):
                b.append(k)
        
        if(len(b) % 2 == 0):
            c = c + k
            b = [] #값 초기화
        else:
            c = c - k
            b = []
    
    return c

File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    let arr = [];
    while(left <= right) {
        arr.push(left);
        left ++;
    }
    return arr.reduce((acc, cur) => {
        let temp = [];
        for (let i = cur; i >= 1; i --) {
            if(cur % i == 0) {
                temp.push(i);
            }
        }
        if(temp.length % 2 == 0) {
            return acc + cur;
        }
        if(temp.length % 2 != 0) {
            return acc - cur;
        }
    }, 0);
}